libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts

```diff
@@ -77,6 +75,88 @@ const getEmployer = (
     : extractAnswer(application.answers, 'employer.nationalRegistryId'),
 })
 
+export const getOtherParentId = (application: Application): string => {
+  const otherParent = extractAnswer<string>(application.answers, 'otherParent')
+  const otherParentId = extractAnswer<string>(
+    application.answers,
+    'otherParentId',
+    '',
+  )
+
+  if (otherParent === formConstants.spouseSelection.spouse) {
+    const familyMembers: NationalRegistryFamilyMember[] | null = extractAnswer(
+      application.externalData,
+      'family.data',
+      null,
+    )
+
+    if (familyMembers === null) {
+      throw new Error(
+        'transformApplicationToParentalLeaveDTO: Cannot find spouse. Missing data for family members.',
+      )
+    }
+
+    const spouse = familyMembers.find(
+      (member) =>
+        member.familyRelation === formConstants.spouseSelection.spouse,
+    )
+
+    if (!spouse) {
+      throw new Error(
+        'transformApplicationToParentalLeaveDTO: Cannot find spouse. No family member with this relation.',
+      )
+    }
+
+    return spouse.nationalId
+  }
+
+  return otherParentId
```

Comment:
Is it fine to return an empty string if nothing can be found?

Reply:
Yes - I think so.

The otherParentId is required in the API payload but optional in the application since a primary parent is not required to enter the national registry id of the other parent.